pkg/clients/resourcerecordset/resourcerecordset.go

```diff
@@ -0,0 +1,177 @@
+
+	rType := string(r.Type)
+	resourceRecords := make([]*v1alpha3.ResourceRecord, 0)
+	for _, item := range r.ResourceRecords {
+		resourceRecords = append(resourceRecords, &v1alpha3.ResourceRecord{Value: item.Value})
+	}
+
+	o := v1alpha3.ResourceRecordSetObservation{
+		Name:            r.Name,
+		Type:            &rType,
+		TTL:             r.TTL,
+		ResourceRecords: resourceRecords,
+	}
+
+	return o
+}
+
+// IsUpToDate checks if object is up to date
+func IsUpToDate(p v1alpha3.ResourceRecordSetParameters, rrset route53.ResourceRecordSet) (bool, error) {
+	patch, err := CreatePatch(&rrset, &p)
+	if err != nil {
+		return false, err
+	}
+	return cmp.Equal(&v1alpha3.ResourceRecordSetParameters{}, patch), nil
+}
+
+// LateInitialize fills the empty fields in *v1alpha3.ResourceRecordSetParameters with
+// the values seen in route53.ResourceRecordSet.
+func LateInitialize(in *v1alpha3.ResourceRecordSetParameters, rrSet *route53.ResourceRecordSet) {
+	if rrSet == nil {
+		return
+	}
+
+	in.Name = awsclients.LateInitializeStringPtr(in.Name, rrSet.Name)
+
+	rrType := string(rrSet.Type)
+	in.Type = awsclients.LateInitializeStringPtr(in.Type, &rrType)
+
+	in.TTL = awsclients.LateInitializeInt64Ptr(in.TTL, rrSet.TTL)
+	if len(in.Records) == 0 && len(rrSet.ResourceRecords) != 0 {
+		in.Records = make([]string, len(rrSet.ResourceRecords))
+		for i, val := range rrSet.ResourceRecords {
+			in.Records[i] = aws.StringValue(val.Value)
+		}
+	}
+}
+
+// CreatePatch creates a *v1beta1.ResourceRecordSetParameters that has only the changed
+// values between the target *v1beta1.ResourceRecordSetParameters and the current
+// *route53.ResourceRecordSet
+func CreatePatch(in *route53.ResourceRecordSet,
+	target *v1alpha3.ResourceRecordSetParameters) (*v1alpha3.ResourceRecordSetParameters, error) {
+	currentParams := &v1alpha3.ResourceRecordSetParameters{}
+	LateInitialize(currentParams, in)
+
+	jsonPatch, err := awsclients.CreateJSONPatch(currentParams, target)
+
+	if err != nil {
+		return nil, err
+	}
+	patch := &v1alpha3.ResourceRecordSetParameters{}
+	if err := json.Unmarshal(jsonPatch, patch); err != nil {
+		return nil, err
+	}
+	patch.ZoneID = patch.Name
+	return patch, nil
+}
+
+// GetResourceRecordSetOrErr returns recordSet if present or err
+func GetResourceRecordSetOrErr(ctx context.Context,
+	c Client, p v1alpha3.ResourceRecordSetParameters) (route53.ResourceRecordSet, error) {
+	req := c.ListResourceRecordSetsRequest(&route53.ListResourceRecordSetsInput{
+		HostedZoneId: p.ZoneID,
+	})
+	res, err := req.Send(ctx)
+	if err != nil {
+		return route53.ResourceRecordSet{}, errors.New(err.Error())
+	}
+
+	for _, rrSet := range res.ResourceRecordSets {
+		if aws.StringValue(rrSet.Name) == *p.Name {
+			return rrSet, nil
+		}
+	}
+
+	return route53.ResourceRecordSet{}, &NotFound{}
+}
+
+// IsRRSetNotFoundErr returns true if the error is because the item doesn't exist
+func IsRRSetNotFoundErr(err error) bool {
```

Comment:
https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

The above article makes a good case for assertion that an error satisfies a particular interface, rather than that it is of a particular concrete type.

Reply:
The requested change to the way `IsRRSetNotFound` error was handled has been changed as per the recommendation from above article.